packages/npm-dependencies-validation/src/depIssuesFinder.ts

```diff
@@ -161,3 +185,95 @@ async function validateMismatchDep(opts: {
     return NO_ISSUES;
   }
 }
+
+async function isMonorepo(absPath: string): Promise<boolean> {
+  const packageJsonPath = resolve(absPath, "package.json");
+  const lernaJsonPath = resolve(absPath, "lerna.json");
+  const pnpmWorkspaceYamlPath = resolve(absPath, "pnpm-workspace.yaml");
+
+  // Check for lerna.json or pnpm-workspace.yaml
```

Comment:
its not enough to check for `lerna.json` or `pnpm-workspace.yaml` in some parent.
The sub-package from which we have arrived must also be included inside the workspace definition...

This is another indication the logical flow to validate this should first "understand" the whole project structure
and only then perform the validations for all the package.json files.